libs/constrained-generators/src/Constrained/Test.hs

```diff
@@ -461,13 +462,15 @@ notSubset :: Spec Fn (Set Int)
 notSubset = constrained $ \s ->
   not_ $ s `subset_` lit (Set.fromList [1, 2, 3])
 
+{- HELP ME  this does not type check
 unionSized :: Spec Fn (Set Int)
 unionSized = constrained $ \s ->
-  10 ==. size_ (s <> lit (Set.fromList [1 .. 8]))
+  Lit (MkSize 10) ==. sizeOf_ @(Set Int) @Fn (s <> lit (Set.fromList [1 .. 8]))
+-}
 
 listSum :: Numbery a => Spec Fn [a]
 listSum = constrained $ \as ->
-  10 <=. sum_ as
+  Lit 10 <=. sum_ as
```

Comment:
Needing `Lit` here is evidence that the interface has gotten worse. This is probably caused by the `OVERLAPPABLE` and `INCOHERENT` instances for `Num Term`, I would prefer if we didn't do this.